ducktape/tests/result.py

```diff
@@ -69,59 +81,141 @@ def __repr__(self):
         return "<%s - test_status:%s, data:%s>" % (self.__class__.__name__, self.test_status, str(self.data))
 
     @property
-    def run_time(self):
+    def run_time_seconds(self):
         if self.start_time < 0:
             return -1
         if self.stop_time < 0:
             return time.time() - self.start_time
 
         return self.stop_time - self.start_time
 
-
-class TestResults(list):
+    def report(self):
+        if not os.path.exists(self.results_dir):
+            mkdir_p(self.results_dir)
+
+        self.dump_json()
+        test_reporter = SingleResultFileReporter(self)
+        test_reporter.report()
+
+    def dump_json(self):
+        """Dump this object as json to the given location. By default, dump into self.results_dir/report.json"""
+        with open(os.path.join(self.results_dir, "report.json"), "w") as fd:
+            json.dump(self, fd, cls=DucktapeJSONEncoder, sort_keys=True, indent=2)
+
+    def to_json(self):
```

Comment:
Meh, I don't have a strong opinion, but I'm weakly more in favor of the way it is now so I don't have to mess with the `@property` fields.

Pros of __dict__:
Maybe easier to keep json representation up to date

Cons of __dict__:
- This `__dict__` won't include `@property` fields, so there'd be some hoops to jump through to get the fields we want
- Current form is nice and explicit and decouples somewhat the internal representation from the json representation